internal/mptrie/store/update_and_query.go

```diff
@@ -118,7 +120,6 @@ func (s *Store) PersistNode(nodePtr []byte) (bool, error) {
 	nodeBytesWithType, ok := s.inMemoryNodes[key]
 	if ok {
 		s.nodesToPersist[key] = nodeBytesWithType
-		delete(s.inMemoryNodes, key)
```

Comment:
why should we remove this line?

Reply:
It doesn't affect anything - we will clear this map in `CommitChanges` call.